pupil_src/launchables/player.py

```diff
@@ -515,12 +508,16 @@ def handle_notifications(n):
                     for p in g_pool.plugins:
                         p.on_char(char_)
 
+                # present frames at appropriate speed
+                g_pool.seek_control.wait(events['frame'].timestamp)
                 glfw.glfwSwapBuffers(main_window)
 
-            # present frames at appropriate speed
-            g_pool.capture.wait(frame)
             glfw.glfwPollEvents()
 
+        # Remove manually to avoid being included in `loaded_plugins`
+        # and `deinit_ui()` call on `g_pool.plugins.clean()`
+        g_pool.plugins._plugins.remove(g_pool.capture)
+
```

Comment:
is the proper way not to raise a notimplemented error in init_gui?

Reply:
The important part is the second line of the comment. `deinit_ui()` causes an exception

Reply:
Ah I misread your comment. Calling deinit_ui() would be legit if init_ui() would have been called before. But we do not do that in Player to avoid having the plugin icon in the side menu. Either we add a switch-case into init_ui and handle it as a normal plugin, or we handle it as a special plugin. But we need to init and add it manually either way. Therefore I would argue to keep this special case.

Reply:
@papr  since the video_playback source is a plugin whats the harm in having a menu. We could add some info like number of frames, current frame, resolution camera name etc...

Reply:
the more we unify the behaviour the better /